src/lib.rs

```diff
@@ -53,6 +60,29 @@ pub trait TcpStack {
 	fn close(&self, socket: Self::TcpSocket) -> Result<(), Self::Error>;
 }
 
+/// This trait is implemented by TCP/IP stacks that expose TCP server functionality. TCP servers
+/// may listen for connection requests to establish multiple unique TCP connections with various
+/// clients.
+pub trait TcpServer: TcpClient {
+	/// Create a new TCP socket and bind it to the specified local port.
+	///
+	///Returns `Ok(socket)` when a new socket is successfully created and bound to the specified
+	///local port. Otherwise, an `Err(e)` variant is returned.
+	fn bind(&self, local_port: u16) -> Result<Self::TcpSocket, Self::Error>;
+
+	/// Begin listening for connection requests on a previously-bound socket.
+	///
+	/// Returns `Ok` if the socket was successfully transitioned to the listening state. Otherwise,
+	/// an `Err(e)` variant is returned.
+	fn listen(&self, socket: &mut Self::TcpSocket) -> Result<(), Self::Error>;
+
+	/// Accept an active connection request on a listening socket.
+	///
+	/// Returns `Ok(connection)` if a new connection was created. If no pending connections are
+	/// available, this function should return [`nb::Error::WouldBlock`].
```

Comment:
Should also return `WouldBlock` if there are no available sockets to use.  I.e. W5500 only has 8, if they're all busy then the connection cannot be accepted.

Reply:
I would imagine that this should actually be an `Error` returned by the network stack instead. Simply waiting longer would not necessarily result in a free socket in the future as there's no guarantee that one of the other users will ever release their socket.

Reply:
That makes sense